File: src/api/scanRules.ts
/**
 * 扫描规则 API 封装
 * Scan rules API wrapper
 */
import { invoke } from '@tauri-apps/api/core'

export interface ScanRule {
  ruleId: string
  threatType: string
  threatName?: string
  provider: string
  op: string
  targetContains?: string
  targetPrefix?: string
  targetPatterns?: string[]
  severity: number
  description?: string
}

export async function loadScanRules(): Promise<ScanRule[]> {
  return await invoke('load_scan_rules')
}

/** MITRE ATT&CK 映射项 / MITRE ATT&CK mapping entry */
export interface MitreRule {
  provider: string
  op: string
  tactic: string
  techniqueId: string
  techniqueName: string
}

/**
 * 函数名称：loadMitreRules
 * 函数作用：从 config/app.json 加载 MITRE ATT&CK 映射规则。
 * Purpose: Loads MITRE ATT&CK mapping rules from config/app.json.
 * 调用方：BehaviorLifecyclePage (行为生命周期页)
 * Called by: BehaviorLifecyclePage (behavior lifecycle page)
 * 中文关键词：MITRE规则，ATT&CK映射，威胁情报，配置加载
 * English keywords: MITRE rules, ATT&CK mapping, threat intelligence, config loading
 */
export async function loadMitreRules(): Promise<MitreRule[]> {
  const config = await invoke<{ enabled: boolean; rules: MitreRule[] }>('load_mitre_rules')
  return config?.rules || []
}

File: src-tauri/src/commands/scan_rules.rs
// 扫描规则命令 — 加载 JSON 规则文件
// Scan rules commands — load JSON rule files
use std::fs;
use std::path::PathBuf;

/// 函数名称：load_scan_rules
/// 函数作用：从 config/scan_rules.json 加载扫描规则列表。
/// Purpose: Loads scan rules list from config/scan_rules.json.
/// Returns: 规则 JSON 数组 / Rule JSON array
/// 调用方：前端初始化
/// 中文关键词：扫描规则，规则加载，JSON配置
/// English keywords: scan rules, rule loading, JSON config
#[tauri::command]
pub fn load_scan_rules() -> Result<serde_json::Value, String> {
    let path = PathBuf::from("config/scan_rules.json");
    if !path.exists() {
        return Ok(serde_json::json!([]));
    }
    let content = fs::read_to_string(&path).map_err(|e| format!("读取扫描规则文件失败: {}", e))?;
    serde_json::from_str(&content).map_err(|e| format!("解析扫描规则失败: {}", e))
}

/// 函数名称：load_mitre_rules
/// 函数作用：从 config/app.json 加载 MITRE ATT&CK 映射规则。
/// Purpose: Loads MITRE ATT&CK mapping rules from config/app.json.
/// Returns: MITRE 规则 JSON / MITRE rules JSON
/// 调用方：前端 BehaviorLifecyclePage
/// 中文关键词：MITRE规则，ATT&CK映射，威胁情报
/// English keywords: MITRE rules, ATT&CK mapping, threat intelligence
#[tauri::command]
pub fn load_mitre_rules() -> Result<serde_json::Value, String> {
    let path = PathBuf::from("config/app.json");
    let content = fs::read_to_string(&path).map_err(|e| format!("读取配置文件失败: {}", e))?;
    let config: serde_json::Value =
        serde_json::from_str(&content).map_err(|e| format!("解析配置文件失败: {}", e))?;

    let mitre = config
        .get("behaviorMitre")
        .cloned()
        .unwrap_or(serde_json::json!({
            "enabled": true,
            "tactics": [],
            "rules": [],
        }));
    Ok(mitre)
}
